ddocs/medic-client/views/messages_by_contact_date/map.js

```diff
@@ -12,9 +12,12 @@ function(doc) {
   };
 
   var emitMessage = function(doc, contact, phone, message) {
-    var id = (contact && contact._id) || phone;
+    var from = (contact && contact._id) || phone;
+    var id = from || doc._id;
     emit([ id, doc.reported_date ], {
       id: doc._id,
+      key: id,
+      from: from,
```

Comment:
We're emitting duplicate data here and I think we really want to keep the value small. Can you just emit the value...

```js
{
  id: doc._id,
  from: (contact && contact._id) || phone,
  ...
}
```

And then in the UI you can specify `value.from || value.id` to get the same result with a smaller view?

Reply:
Yeah definitely. I had run out of brain waves when I wrote this PR and couldn't work out what to change in the front end to not require all these things. If I give it a once over with a fresh mind I'm sure we can clean a lot of this stuff up (including, for example, not emitting the `id` because it's already part of the view key, and perhaps renaming things like "key" into something less obtuse)